<?xml version="1.0"?>
<robot name="pupper_leg">

  <!-- Materials -->

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <material name="orange">
    <color rgba="1 0.3569 0 1"/>
  </material>  

  <!-- Joints -->

  <joint name="world_to_fixture" type="fixed">
    <parent link="world"/>
    <child link="fixture"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>

  <joint name="fixture_to_base_link" type="continuous">
    <parent link="fixture"/>
    <child link="base_link"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0 0 0.10"/>
  </joint>

  <joint name="base_link_to_upper_leg" type="continuous">
    <parent link="base_link"/>
    <child link="upper_leg"/>
    <axis xyz="0 0 1"/>
    <origin rpy="1.5708 0 0" xyz="0 0 0.028"/>
  </joint>

  <joint name="upper_leg_to_lower_leg" type="continuous">
    <parent link="upper_leg"/>
    <child link="lower_leg"/>
    <axis xyz="0 0 1"/>
    <origin rpy="3.14159 0 0" xyz="-0.08004 0 0.040"/>
  </joint>

  <joint name="lower_leg_to_end_effector" type="fixed">
    <parent link="lower_leg"/>
    <child link="end_effector"/>
    <origin rpy="0 0 0" xyz="-0.11003 0 0.006"/>
  </joint>

  <!-- Links -->

  <link name="world"/>

  <link name="fixture">
    <visual>
      <geometry>
        <cylinder length="0.10" radius="0.025"/>     
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.10" radius="0.025"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
    </collision>
    <inertial>
      <mass value="2.0"/>
      <inertia ixx="0.001979" ixy="0" ixz="0" iyy="0.001979" iyz="0" izz="0.0001041"/>	
    </inertial>	    
  </link>

  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/hip_rotated.stl"
         scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 4.71239" xyz="0 0 0"/>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/hip_rotated.stl"
         scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="0 0 4.71239" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="0.0003646" ixy="0" ixz="0" iyy="0.0003646" iyz="0" izz="0.00005208"/>
    </inertial>	    
  </link>

  <link name ="upper_leg">
    <visual>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/upper_leg.stl"
         scale="0.001 0.001 0.001"/> 
      </geometry>
      <origin rpy="3.14159 -1.57080 0" xyz="0 0 0.003"/>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/upper_leg.stl"
         scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="3.14159 -1.57080 0" xyz="0 0 0.003"/>
    </collision>
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="0.006094" ixy="0" ixz="0" iyy="0.006094" iyz="0" izz="0.0001563"/>
    </inertial>	    
  </link>

  <link name ="lower_leg">
    <visual>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/lower_leg.stl"
         scale="0.001 0.001 0.001"/>        
      </geometry>
      <origin rpy="-3.14159 -1.57080 0" xyz="0 0 0.004"/>
      <material name="white"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="file:///home/ubuntu/robotic_arms_ws/src/bazu/src/urdf/lower_leg.stl"
         scale="0.001 0.001 0.001"/>
      </geometry>
      <origin rpy="-3.14159 -1.57080 0" xyz="0 0 0.004"/>
    </collision>	      
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="0.006094" ixy="0" ixz="0" iyy="0.006094" iyz="0" izz="0.0001563"/>	    
    </inertial>	    
  </link>

  <link name ="end_effector">
    <visual>
      <geometry>
        <sphere radius="0.002"/>
      </geometry>
      <origin rpy="0 0 0 " xyz="0 0 0"/>
      <material name="orange"/>
    </visual>
    <collision>
      <geometry>
        <sphere radius="0.002"/>
      </geometry>
      <origin rpy="0 0 0 " xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="3.0"/>
      <inertia ixx="0.006094" ixy="0" ixz="0" iyy="0.006094" iyz="0" izz="0.0001563"/>
    </inertial>
  </link>

  <!-- PositionJointInterface -->
  <!-- EffortJointInterface -->

  <transmission name="transmission_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="fixture_to_base_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>	      
  </transmission>

  <transmission name="transmission_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="base_link_to_upper_leg">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="transmission_3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="upper_leg_to_lower_leg">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <gazebo>
    <plugin name="control" filename="libgazebo_ros_control.so"/>
  </gazebo>	  
	  
</robot>
